### ISSUE TITLE
UTF8 std::string instead of icu::UnicodeString?

### ISSUE THREAD

**Issue by springmeyer** [mapnik/mapnik#1985]
Companion to #1813 - ticket to track research into potentially supporting a lightweight build mode that does not use `icu::UnicodeString` and does not support any other encodings other than `utf8`.

**Comment by springmeyer** [comment]
@artemp - We revisit this in light of #3367. I think the clincher is text shaping which depends on ICU. But I still wonder if we could make ICU an optional dep if:

  - We used `std::regex` over `boost::regex with icu support`
  - We offered a mode to disable complex text shaping
  - We dropped support for alternative encodings and only supported utf8 in plugins. (we should probably do this anyway to be modern and simple)

**Comment by lightmare** [comment]
I don't really have much insight into this. Some time ago I too asked "could we not use icu?" and ended up in text shaping like you, where I dropped it and focused on something else ;)

Isn't this a deficiency of ICU that should be fixed upstream? If the point of version-suffixed exported functions is the ability to link against multiple versions of the library, then it shouldn't be calling clashing unversioned exported functions or using incompatible data files in the first place.

**Comment by artemp** [comment]
I agree with @lightmare, if ICU supports versioned export functions it needs fixing ^

**Comment by artemp** [comment]
@springmeyer - only supporting `UTF-8` is neat idea ! As for shaping - we do need to support RTL/LTR shaping so.. maybe time is good to look into this again. Also, an old idea of storing strings as UTF-8 internally comes to mind ..
